mlflow/sklearn/utils.py

```diff
@@ -330,44 +441,80 @@ def _get_regressor_metrics(fitted_estimator, fit_args, fit_kwargs):
 def _log_warning_for_metrics(func_name, func_call, err):
     msg = (
         func_call.__qualname__
-        + " failed. The "
+        + " failed. The metric "
         + func_name
-        + " metric will not be recorded. Metric error: "
+        + "will not be recorded."
+        + " Metric error: "
         + str(err)
     )
     _logger.warning(msg)
 
 
+def _log_warning_for_artifacts(func_name, func_call, err):
+    msg = (
+        func_call.__qualname__
+        + " failed. The artifact "
+        + func_name
+        + "will not be recorded."
+        + " Artifact error: "
+        + str(err)
+    )
+    print("2: ", msg)
+    _logger.warning(msg)
+
+
 def _log_specialized_estimator_content(fitted_estimator, run_id, fit_args, fit_kwargs):
     import sklearn
 
+    mlflow_client = MlflowClient()
     name_metric_dict = {}
     try:
         if sklearn.base.is_classifier(fitted_estimator):
             name_metric_dict = _get_classifier_metrics(fitted_estimator, fit_args, fit_kwargs)
+
         elif sklearn.base.is_regressor(fitted_estimator):
             name_metric_dict = _get_regressor_metrics(fitted_estimator, fit_args, fit_kwargs)
-
     except Exception as err:  # pylint: disable=broad-except
         msg = (
             "Failed to autolog metrics for "
             + fitted_estimator.__class__.__name__
             + ". Logging error: "
             + str(err)
         )
+        print("3: ", msg)
         _logger.warning(msg)
-
     else:
         # batch log all metrics
         try_mlflow_log(
-            MlflowClient().log_batch,
+            mlflow_client.log_batch,
             run_id,
             metrics=[
                 Metric(key=str(key), value=value, timestamp=int(time.time() * 1000), step=0)
                 for key, value in name_metric_dict.items()
             ],
         )
 
+    name_artifact_dict = {}
+    try:
+        if sklearn.base.is_classifier(fitted_estimator):
+            name_artifact_dict = _get_classifier_artifacts(fitted_estimator, fit_args, fit_kwargs)
+    except Exception as err:  # pylint: disable=broad-except
+        msg = (
+            "Failed to autolog artifacts for "
+            + fitted_estimator.__class__.__name__
+            + ". Logging error: "
+            + str(err)
+        )
+        print("4: ", msg)
+        _logger.warning(msg)
+    else:
+        if bool(name_artifact_dict):
+            with TempDir() as tmp:
+                for name, display in name_artifact_dict.items():
+                    filepath = os.path.join(tmp.path(), "{}.png".format(name))
```

Comment:
Nit: I think we can simplify this as:

```suggestion
                    filepath = tmp.path("{}.png".format(name))
```